src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java

```diff
@@ -48,14 +62,31 @@ public class IoTCoreClient implements MessageClient {
         }
     };
 
+    public MqttClientConnectionEvents connectionCallbacks = new MqttClientConnectionEvents() {
+        @Override
+        public void onConnectionInterrupted(int errorCode) {
+        }
+
+        @Override
+        public void onConnectionResumed(boolean sessionPresent) {
+            // subscribe to any topics left to be subscribed
+            Set<String> topicsToSubscribe = new HashSet<>(toSubscribeIotCoreTopics);
+            topicsToSubscribe.removeAll(subscribedIotCoreTopics);
+            subscribeFuture = executorService.submit(() -> subscribeToTopicsWithRetry(topicsToSubscribe));
```

Comment:
kill existing future, if any.

Reply:
Using the future must be synchronized

Reply:
We cancel this future before starting with subscribe: [link](https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/blob/56fd72b2182f65fb710644c93429f89c3c94795a/src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java#L143-L145). 
All `future` updates are within synchronized function, except for during onConnectionResumed [here](https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/blob/56fd72b2182f65fb710644c93429f89c3c94795a/src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java#L75). Let me synchronize that too.

Reply:
updated, thanks.